packages/edit-site/src/components/page-pages/index.js

```diff
@@ -326,11 +332,64 @@ export default function PagePages() {
 				header: __( 'Date' ),
 				id: 'date',
 				render: ( { item } ) => {
-					const formattedDate = dateI18n(
-						getSettings().formats.datetimeAbbreviated,
-						getDate( item.date )
+					const isDraftOrPrivate = [ 'draft', 'private' ].some(
+						( status ) => status === item.status
 					);
-					return <time>{ formattedDate }</time>;
+					if ( isDraftOrPrivate ) {
+						return (
+							<>
+								{ __( 'Modified: ' ) }
+								<time>{ getFormattedDate( item.date ) }</time>
+							</>
+						);
+					}
+
+					const isPending = item.status === 'pending';
+					if ( isPending ) {
+						const isModified =
+							getDate( item.modified ) > getDate( item.date );
+						const dateToDisplay = isModified
+							? item.modified
+							: item.date;
+						return (
+							<>
```

Comment:
I think we need to use `createInterpolateElement` here.. cc @mcsf

Reply:
Thanks for bringing it up. I was looking into that but had issues with collapsed whitespace, note how the `Scheduled:May` here:

<img width="377" alt="Captura de ecrã 2024-05-16, às 13 34 32" src="https://github.com/WordPress/gutenberg/assets/583546/58334567-4ff3-4cdd-b71c-37e0ca795b48">

It turns out it wasn't the interpolation but a CSS rule (`display: flex`) of the `dataviews-view-table__cell-content-wrapper`. I just figured the text node was considered a different node and its whitespace collapsed, so I added a `span` wrapper for the text node & the time to be considered a single element by its flex container.